flang/lib/Lower/ConvertProcedureDesignator.cpp

```diff
@@ -98,6 +99,10 @@ hlfir::EntityWithAttributes Fortran::lower::convertProcedureDesignatorToHLFIR(
     mlir::Location loc, Fortran::lower::AbstractConverter &converter,
     const Fortran::evaluate::ProcedureDesignator &proc,
     Fortran::lower::SymMap &symMap, Fortran::lower::StatementContext &stmtCtx) {
+  if (std::optional<fir::FortranVariableOpInterface> varDef =
+          symMap.lookupVariableDefinition(*proc.GetSymbol()))
```

Comment:
Actually, you should only do this bypass if `proc.GetSymbol()` is not a nullptr (it may be if the procedure reference is an unrestricted intrinsic).

e.g. the following will crash with HLFIR enabled after this patch:

```
  intrinsic iabs
  call test(iabs)
end
```

Reply:
Accepted. Good catch!